### PR TITLE
Add duplicated-argument-name rule

### LINKED ISSUE #401: [Rule] Duplicate argument name
Duplicating argument name should be detected. In this example:
```
*** Test Cases ***
Test keyword with duplicate arg
    Keyword    1    2


*** Keywords ***
Keyword
    [Arguments]    ${var}    ${var}
    Log  ${var}
```
Robocop doesn't warn about duplicate argument. When running the code Robot Framework also doesn't complain and just overwrite first ${var} with second one:
image

### LINKED ISSUE #480: [Rule] Duplicated assign variable name
```
${var}  ${var}  ${Var}   My Keyword
```

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #401 
Closes #480

### CODE REVIEW

File: robocop/checkers/duplications.py [hunk 1]
```diff
@@ -48,6 +48,11 @@ class DuplicationsChecker(VisitorChecker):
             "duplicated-variables-import",
             'Duplicated variables import with path "%s" in suite',
             RuleSeverity.WARNING
+        ),
+        "0811": (
+            "duplicated-argument-name",
+            "Argument name '%s' was already used",
```
Comment (mnojek) [inline robocop/checkers/duplications.py:54]:
```suggestion
            "Argument name '%s' is already used",
```

File: tests/atest/rules/duplications/duplicated-argument-name/expected_output.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+${rules_dir}${\}test.robot:3:30 [E] 0811 Argument name '${var}' was already used
+${rules_dir}${\}test.robot:4:12 [E] 0811 Argument name '${var}' was already used
+${rules_dir}${\}test.robot:4:22 [E] 0811 Argument name '${Var}' was already used
+${rules_dir}${\}test.robot:4:32 [E] 0811 Argument name '${var}' was already used
```
Comment (mnojek) [inline tests/atest/rules/duplications/duplicated-argument-name/expected_output.txt:4]:
```suggestion
${rules_dir}${\}test.robot:3:30 [E] 0811 Argument name '${var}' is already used
${rules_dir}${\}test.robot:4:12 [E] 0811 Argument name '${var}' is already used
${rules_dir}${\}test.robot:4:22 [E] 0811 Argument name '${Var}' is already used
${rules_dir}${\}test.robot:4:32 [E] 0811 Argument name '${var}' is already used
```

**Comment by mnojek** [comment]
What about assignments from keyword calls?

**Comment by bhirsz** [comment]
> What about assignments from keyword calls?

Good idea, I will create separate rule for it.

File: robocop/checkers/duplications.py [hunk 3]
```diff
@@ -48,6 +48,16 @@ class DuplicationsChecker(VisitorChecker):
...
             RuleSeverity.WARNING
+        ),
+        "0811": (
+            "duplicated-argument-name",
+            "Argument name '%s' is already used",
+            RuleSeverity.ERROR
+        ),
+        "0812": (
+            "duplicated-assign-name",
+            "Assign variable name '%s' is already used",
```
Comment (mnojek) [inline robocop/checkers/duplications.py:59]:
```suggestion
            "Assigned variable name '%s' is already used",
```

File: robocop/checkers/duplications.py [hunk 4]
```diff
@@ -48,6 +48,16 @@ class DuplicationsChecker(VisitorChecker):
...
             'Duplicated variables import with path "%s" in suite',
             RuleSeverity.WARNING
+        ),
+        "0811": (
+            "duplicated-argument-name",
+            "Argument name '%s' is already used",
+            RuleSeverity.ERROR
+        ),
+        "0812": (
+            "duplicated-assign-name",
```
Comment (mnojek) [inline robocop/checkers/duplications.py:58]:
```suggestion
            "duplicated-assigned-var-name",
```
?

File: robocop/checkers/duplications.py [hunk 5]
```diff
@@ -88,10 +98,24 @@ def check_duplicates(self, container, rule):
...
 
     def visit_Keyword(self, node):  # noqa
         keyword_name = normalize_robot_name(node.name)
         self.keywords[keyword_name].append(node)
+        self.generic_visit(node)
+
+    def visit_KeywordCall(self, node):  # noqa
+        assign = node.get_tokens(Token.ASSIGN)
+        if not assign:
+            return
```
Comment (mnojek) [inline robocop/checkers/duplications.py:111]: I think this is not needed because if it is empty, then the for loop below will not be executed

**Review by mnojek — APPROVED** [review]

**Pull request merged**